Realm/RLMManagedArray.mm

```diff
@@ -198,9 +198,19 @@ - (NSUInteger)countByEnumeratingWithState:(NSFastEnumerationState *)state
 }
 
 - (id)objectAtIndex:(NSUInteger)index {
-    return translateErrors([&] {
-        RLMAccessorContext context(*_objectInfo);
-        return _backingList.get(context, index);
+    return translateErrors([&]() -> id {
+        realm::Mixed mixed = _backingList.get_any(index);
+        if (mixed.is_type(realm::type_Dictionary)) {
+            return [[RLMManagedDictionary alloc] initWithBackingCollection:_backingList.get_dictionary(index) parentInfo:_ownerInfo property:_property];
+        }
+        else if (mixed.is_type(realm::type_List)) {
+            return [[RLMManagedArray alloc] initWithBackingCollection:_backingList.get_list(index) parentInfo:_ownerInfo property:_property];
+        }
```

Comment:
This is all handled by `List::get()` and doesn't need to be special-cased here or in RLMManagedDictionary.

Reply:
Are you sure that's the case, I was instructed by @jedelbo that this was the api for getting nested collections

Reply:
All of the tests pass with both of these just calling get() and there's code in get() which does exactly this.